src/Support/ExceptionHandler.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Spatie\Ray\Support;
+
+use Spatie\Ray\Ray;
+
+class ExceptionHandler
+{
+    public function catch(Ray $ray, $callback): Ray
+    {
+        $this->processCallback($ray, $callback);
+
+        if (! empty(Ray::$caughtExceptions)) {
+            throw array_shift(Ray::$caughtExceptions);
+        }
+
+        return $ray;
+    }
+
+    protected function handleCallable(Ray $ray, $callback, $rethrow = true): Ray
+    {
+        $paramType = $this->getParamType(new \ReflectionFunction($callback));
+        $expectedClasses = $this->getExpectedClasses($paramType);
+
+        if (count($expectedClasses)) {
+            $isExpected = false;
+
+            foreach ($expectedClasses as $expectedClass) {
+                foreach (Ray::$caughtExceptions as $caughtException) {
+                    if (is_a($caughtException, $expectedClass, true)) {
+                        $isExpected = true;
+
+                        break 2;
```

Comment:
Could we rewrite this to avoid having that `break 2;` Maybe a function that loops through these things and returns a boolean that determines if we should continue